src/main/Core/ImageGenerator/LinuxIconExtractor.ts

```diff
@@ -0,0 +1,412 @@
+import type { CommandlineUtility } from "@Core/CommandlineUtility";
+import type { FileSystemUtility } from "@Core/FileSystemUtility";
+import type { IniFileParser } from "@Core/IniFileParser/IniFileParser";
+import type { Logger } from "@Core/Logger";
+import type { Image } from "@common/Core/Image";
+import { dirname, extname, join } from "path";
+import type { CacheFileNameGenerator } from "./CacheFileNameGenerator";
+import type { FileIconExtractor } from "./FileIconExtractor";
+
+// Specs from https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html
+type IconThemeSubdir = {
+    Size: number;
+    Scale: number; // Defaults to 1
+    MinSize: number; // Defaults to size
+    MaxSize: number; // Defaults to size
+    Threshold: number; // Defaults to 2
+    Type: "Fixed" | "Scalable" | "Threshold"; // Defaults to Threshold
+};
+
+type IconTheme = {
+    name: string;
+    path: string;
+    // Paths in order as listed in index.theme
+    subdirectories: string[];
+    // Map<subdirPath, subdirData>
+    subdirData: Map<string, IconThemeSubdir>;
+    parents: string[];
+};
+
+export class LinuxIconExtractor implements FileIconExtractor {
+    private searchCache: Map<string, IconTheme[]>;
+    private userTheme: string;
+    private readonly baseDirectories: string[];
+    private readonly folderPaths: Record<string, string>;
+
+    public constructor(
+        private readonly fileSystemUtility: FileSystemUtility,
+        private readonly commandlineUtility: CommandlineUtility,
+        private readonly iniFileParser: IniFileParser,
+        private readonly logger: Logger,
+        private readonly cacheFileNameGenerator: CacheFileNameGenerator,
+        private readonly cacheFolder: string,
+        private readonly homePath: string,
+    ) {
+        this.baseDirectories = [
+            join(this.homePath, ".icons"),
+            join(process.env["XDG_DATA_HOME"] || join(this.homePath, ".local", "share"), "icons"),
```

Comment:
By the way, we can now use the new `EnvironmentVariableProvider` here instead of reyling on `process.env`.  This should make it easier to mock it in unit tests.